src/TimeGrid.jsx

```diff
@@ -77,9 +80,24 @@ export default class TimeGrid extends Component {
       this.measureGutter()
     }
 
+    if (this._scrollRatio) {
+      const { content } = this.refs;
+      content.scrollTop = content.scrollHeight * this._scrollRatio;
+      // Only do this once
+      this._scrollRatio = null;
+    }
+
     //this.checkOverflow()
   }
 
+  componentWillReceiveProps(nextProps) {
+    const { start, scrollTop } = this.props;
+    // When paginating, reset scroll
+    if (!dates.eq(nextProps.start, start) || nextProps.scrollTop !== scrollTop) {
```

Comment:
`scrollTop` wrong, yes. The condition, however is correct, because it only calculates the new scroll value when the start is different (i.e. it has changed dates or paginated) or when the `scrollToTime` prop has changed.